src/legacy/core_plugins/metrics/index.ts

```diff
@@ -18,34 +18,36 @@
  */
 
 import { resolve } from 'path';
+import { Legacy } from 'kibana';
+import { PluginInitializerContext } from 'src/core/server';
+import { InternalCoreSetup } from 'src/core/server';
+import { metricsPlugin } from './server/np/';
 
-import { fieldsRoutes } from './server/routes/fields';
-import { visDataRoutes } from './server/routes/vis';
-import { SearchStrategiesRegister } from './server/lib/search_strategies/search_strategies_register';
-
-export default function(kibana) {
+// eslint-disable-next-line import/no-default-export
+export default function(kibana: any) {
   return new kibana.Plugin({
+    id: 'metrics',
+
     require: ['kibana', 'elasticsearch'],
```

Comment:
To be safe, you'll want to add `visualizations` to this list. This isn't a new platform thing, but useful for the legacy world: it's a safeguard that causes kibana to throw an error if for some reason one of these required plugins is disabled.

Also probably add `data`, since it will eventually be a dependency of this plugin too.

Reply:
`visualizations` and `data` are added.